clusterloader2/pkg/prometheus/prometheus.go

```diff
@@ -183,6 +194,44 @@ func (pc *PrometheusController) exposeKubemarkApiServerMetrics() error {
 	return nil
 }
 
+// runNodeExporter adds node-exporter as master's static manifest pod.
+func (pc *PrometheusController) runNodeExporter() error {
+	klog.Infof("Starting node-exporter on master nodes.")
+	kubemarkFramework, err := framework.NewFramework(&pc.clusterLoaderConfig.ClusterConfig, numK8sClients)
+	if err != nil {
+		return err
+	}
+
+	// Validate masters first
+	nodes, err := client.ListNodes(kubemarkFramework.GetClientSets().GetClient())
+	if err != nil {
+		return err
+	}
+
+	var g errgroup.Group
+	numMasters := 0
+	for _, node := range nodes {
+		node := node
+		if system.IsMasterNode(node.Name) {
+			numMasters++
+			g.Go(func() error {
+				f, err := os.Open(os.ExpandEnv(nodeExporterPod))
+				if err != nil {
+					return fmt.Errorf("Unable to open manifest file: %v", err)
+				}
+				defer f.Close()
+				return util.SSH("sudo tee /etc/kubernetes/manifests/node-exporter.yaml > /dev/null", &node, f)
```

Comment:
I'm not a huge fan of this approach - can we at least add a TODO to figure out if we can do that in a better way in the future?

Reply:
Added TODO.